sdk/core/azure-core/src/main/java/com/azure/core/implementation/http/rest/SwaggerMethodParser.java

```diff
@@ -470,9 +478,11 @@ public UnexpectedExceptionInformation getUnexpectedException(int code) {
      * Get the object to be used as the value of the HTTP request.
      *
      * @param swaggerMethodArguments the method arguments to get the value object from
+     * @param serializer {@link SerializerAdapter} used to encode the request body if it's an
+     * {@code application/x-www-form-urlencoded} request.
      * @return the object that will be used as the body of the HTTP request
      */
-    public Object setBody(Object[] swaggerMethodArguments) {
+    public Object setBody(Object[] swaggerMethodArguments, SerializerAdapter serializer) {
```

Comment:
Do we need to pass the serializerAdapter for every method? If so, should we just make this a param in `getSwaggerInterfaceParser` like we had before in the ctor?

Reply:
No, `SerializerAdapter` isn't needed for each method, nor does it exist in `SwaggerInterfaceParser` anymore